### PR TITLE
feat(angular): Upgrade to Angular 5

### PR DESCRIPTION
**Pull request by DavidRouyer**
- Update dependencies
- Remove unnecessary JSDoc tags
- Fix build process

BREAKING CHANGE: The upgrade to Angular 5 breaks compatibility with Angular 4.

### CODE REVIEW

File: src/lib/src/services/notifier-timer.service.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export class NotifierTimerService {
 	/**
 	 * Timeout ID, used for clearing the timeout later on
 	 */
-	private timerId: number;
+	private timer: NodeJS.Timer;
```
Comment (dominique-mueller) [inline src/lib/src/services/notifier-timer.service.ts]: I think this is an issue introduced by the NodeJS typings, the timer ID here should be of type `number` (as before), not `NodeJS.Timer`. Perhaps we should cast this into a number, if necessary?

File: package.json [hunk 2]
```diff
@@ -48,64 +48,60 @@
...
-	},
 	"devDependencies": {
-		"@angular/common": "4.1.x",
-		"@angular/compiler-cli": "4.1.x",
-		"@angular/compiler": "4.1.x",
-		"@angular/core": "4.1.x",
-		"@angular/platform-browser-dynamic": "4.1.x",
-		"@angular/platform-browser": "4.1.x",
-		"@types/jest": "20.0.x",
+		"@angular/common": "^5.0.3",
```
Comment (dominique-mueller) [inline package.json]: I'd usually prefer `5.0.x` instead of `^5.0.3` as it is easier to understand for developers

File: package.json [hunk 3]
```diff
@@ -48,64 +48,60 @@
 		"ci:coverage": "codecov -f coverage/coverage-final.json",
 		"ci:test": "jest --config tools/jest/jest.config.json --runInBand --no-cache"
 	},
-	"peerDependencies": {
```
Comment (dominique-mueller) [inline package.json]: Please do not remove the peerDependencies, instead update to `>= 5.0.0 < 6.0.0`

File: src/lib/src/services/notifier-timer.service.ts (hunk 1 above)
Comment (DavidRouyer) [reply src/lib/src/services/notifier-timer.service.ts]: You are right, TypeScript was complaining about this, I didn't check the signature of the method setTimeout which is wrong https://github.com/Microsoft/TypeScript/issues/842 I'll use window.setTimeout instead.

**Comment by DavidRouyer** [comment]
@dominique-mueller I've addressed your comments :)

**Comment by dominique-mueller** [comment]
Thanks again @DavidRouyer for your contribution! I will do some further build-related updates and release v3 of angular-notifier in the next few days.

**Pull request merged**